docs/PATCH_RELEASE_TEMPLATE.md

```diff
@@ -0,0 +1,27 @@
+# Patch Release Checklist
+
+This process handles patch releases from version `vX.Y.Z` to `vX.Y.Z+1` assuming that `vX.Y.Z` is the latest released version of go-ipfs.
+
+- [ ] Fork a new branch (`release-vX.Y.Z`) from `release` and cherry-pick the relevant commits from master (or custom fixes) onto this branch
+- [ ] Make a minimal changelog update tracking the relevant fixes to CHANGELOG.
```

Comment:
I was thinking to just make this free hand and look something like https://github.com/ipfs/go-ipfs/pull/8267#issue-691778061 instead of something more formal or using mkreleaselog. Thoughts?